### PR TITLE
Split location schema & document into its own file

### PR DESCRIPTION
**Pull request by khmoran**

### CODE REVIEW

**Comment by khmoran** [comment]
[Separate JS code into classes with some object-oriented practices.](https://trello.com/c/rywyPfXt/92-separate-js-code-into-classes-with-some-object-oriented-practices)

File: data-serving/data-service/src/model/location.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        return (
+            this != null &&
+            this.country == null &&
+            this.administrativeAreaLevel1 == null &&
+            this.administrativeAreaLevel2 == null &&
+            this.locality == null
+        );
+    },
+    'One of location.country, location.administrativeRegion1, ' +
+        'location.administrativeRegion2, or locality is required',
```
Comment (axmb) [inline data-serving/data-service/src/model/location.ts:14]: Which schema do these prefixes refer to (i.e. why isn't locality prefixed with `location.`)?

**Comment by khmoran** [comment]
[Refactor dataserver model into classes, add tests](https://trello.com/c/9VunjXvq/133-refactor-dataserver-model-into-classes-add-tests)

File: data-serving/data-service/src/model/location.ts (hunk 1 above)
Comment (khmoran) [reply data-serving/data-service/src/model/location.ts]: because i am bad at typinggg -- going to remove the prefixes, it's confusing

**Pull request merged**